lib/component.js

```diff
@@ -276,6 +307,9 @@ class Component extends WebComponent {
 
     this.panelID = cuid();
 
+    // to enforce naviagate always point to this class
+    this.navigate = this.navigate.bind(this);
```

Comment:
in case `navigate` need bo be passed in params by some component with routes

Reply:
I suspect that this use case is rare enough that if a component needs to do this, it can use arrow functions by itself instead of binding in literally every component.

Reply:
I still think we should remove this binding.